core/src/main/java/greencity/controller/UserController.java

```diff
@@ -209,9 +210,17 @@ public ResponseEntity<PageableDto<UserForListDto>> getUsersByFilter(
         @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
     })
     @GetMapping
-    public ResponseEntity<UserUpdateDto> getUserByPrincipal(@ApiIgnore @AuthenticationPrincipal Principal principal) {
-        String email = principal.getName();
-        return ResponseEntity.status(HttpStatus.OK).body(userService.getUserUpdateDtoByEmail(email));
+    public ResponseEntity<UserUpdateDto> getUserByPrincipal(@ApiIgnore Principal principal) {
+        if (principal != null){
```

Comment:
The same problem which i described in your other pullRequest:[ [316] fixed response code form 401 UNAUTHORIZED to 403 FORBIDDEN](https://github.com/GreenCityProject/GreenCityUser21_team2/pull/14) .

The logic should do AccessDeniedHandler:
`  if (principal != null){
            try {
                String email = principal.getName();
                return ResponseEntity.status(HttpStatus.OK).body(userService.getUserUpdateDtoByEmail(email));
            }catch (AccessDeniedException e){
                return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
            }
        } else {
            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
        }
`